http/client_acp.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+// AddPolicyResult wraps the result of successfully adding/registering a Policy.
+type AddPolicyRequest struct {
+	// Policy body in JSON or YAML format.
+	Policy string `json:"policy"`
+}
+
+func (c *Client) AddPolicy(
```

Comment:
question: Codecov says this is untested, do you know why?

Reply:
That's odd, I don't have an explicit `client.AddPolicy` test but still this should be hit.

Ah good catch I think I know why.

Reply:
This is likely because in the wrapper I was using `node` for both client and skipped actually piping to `cmd.execute` in cli wrapper and instead just used ``` return w.client.AddPolicy(ctx, creator, policy) ```.

```go
func (w *Wrapper) AddPolicy(
	ctx context.Context,
	creator string,
	policy string,
) (client.AddPolicyResult, error) {
	return w.node.AddPolicy(ctx, creator, policy)
}
```

I think I have implemented it properly now, please give the test wrapper code a look and lmk incase I missed anything.

Reply:
LGTM, thanks Shahzad :)